### ISSUE TITLE
DeprecationWarning on Python 3.7/3.8 and breakage on master/3.10 due to coreapi dependency

### ISSUE THREAD

**Issue by johnthagen** [axnsan12/drf-yasg#473]
Originally posted at https://github.com/encode/django-rest-framework/issues/6991

Relates to #389 
___________

Python 3.8 releases today but this warning is still triggered from `djangorestframework` 3.10.3 and `drf-yasg` 1.17.0 :

https://github.com/tomchristie/itypes/issues/11

This makes causes users who run their DRF testing with `-Werror` fail the build if they also use `drf-yasg`.

From: https://docs.python.org/3/library/collections.html#module-collections

> Deprecated since version 3.3, will be removed in version 3.10: Moved Collections Abstract Base Classes to the collections.abc module.

##  Steps to reproduce

```py
from rest_framework.test import APIClient
```

Run `pytest` tests with`-Werror` under Python 3.7.

```
$ python3 -Werror -m pytest
```

```
During handling of the above exception, another exception occurred:
.tox/docker/lib/python3.7/site-packages/_pytest/config/__init__.py:462: in _importconftest
    mod = conftestpath.pyimport()
.tox/docker/lib/python3.7/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:677: in _load_unlocked
    ???
.tox/docker/lib/python3.7/site-packages/_pytest/assertion/rewrite.py:142: in exec_module
    exec(co, module.__dict__)
rest_api/tests/conftest.py:2: in <module>
    from rest_framework.test import APIClient
.tox/docker/lib/python3.7/site-packages/rest_framework/test.py:16: in <module>
    from rest_framework.compat import coreapi, requests
.tox/docker/lib/python3.7/site-packages/rest_framework/compat.py:98: in <module>
    import coreapi
.tox/docker/lib/python3.7/site-packages/coreapi/__init__.py:2: in <module>
    from coreapi import auth, codecs, exceptions, transports, utils
.tox/docker/lib/python3.7/site-packages/coreapi/codecs/__init__.py:2: in <module>
    from coreapi.codecs.base import BaseCodec
.tox/docker/lib/python3.7/site-packages/coreapi/codecs/base.py:1: in <module>
    import itypes
.tox/docker/lib/python3.7/site-packages/itypes.py:2: in <module>
    from collections import Mapping, Sequence
<frozen importlib._bootstrap>:1032: in _handle_fromlist
    ???
.tox/docker/lib/python3.7/collections/__init__.py:52: in __getattr__
    DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```

**Comment by nathanprat** [comment]
Hello,
Same issue with `drf-yasg==1.17.0`.
Replying to bump this issue.

**Comment by doom3wasagoodgame** [comment]
same here, bumping

**Comment by johnthagen** [comment]
Updated issue title based on updated Python docs stating 3.10 is when this breakage will occur.

**Comment by johnthagen** [comment]
`coreapi` isn't actually used in that many places in this library: https://github.com/axnsan12/drf-yasg/search?q=coreapi&unscoped_q=coreapi

**Comment by johnthagen** [comment]
For anyone else tracking this, it looks like [`itypes` has a new maintainer](https://github.com/PavanTatikonda/itypes) (@PavanTatikonda) and this particular issue has been fixed and released onto PyPI (v1.2.0).

https://github.com/PavanTatikonda/itypes/commit/e9b32979d1eaefc3c08780de496a20520fa677cd
